### DISCUSSION TITLE
How to make `Insert link` unchecked by default when sharing into file

### DISCUSSION THREAD

**Discussion by GlassGruber** [gsantner/markor#2378] [Q&A] [answered]
When sharing to Markor, the check box for `Insert link` is _checked_ by default. I rarely use it, so I have to often uncheck it before selecting the document.  
Is there a setting to make it optin/unchecked by default?

Thank you!

**Comment by harshad1** [comment]
You don't want your links formatted as links?

We could make it a remembered  option.

**Reply by GlassGruber** [reply]
> You don't want your links formatted as links?
> 
> We could make it a remembered option.

Thank you for answering! Yea that would be nice, I mainly use plain old text, when using markdown I mostly leave links raw and let the viewer convert it to a clickable link. 

I find it a bit odd that urls are converted now to `<a>` links elements, it was a bit more sensed IMO in previous versions when it was converted to a MD url using the domain as anchor tag like `[example.com](https://example.com/some-long-url)` and IIRC it was possible to disable this conversion.

**Reply by gsantner** [reply, marked as answer]
@GlassGruber 
The conversion to Markdown syntax is still there, I use that every day and also at the latest release.

You just need to share to a file that is really configured as Markdown file and not something else.

If you have the file not set to Format=Markdown, it can format the link differently. And that seems to be the case here.

**Reply by GlassGruber** [reply]
> You just need to share to a file that is really configured as Markdown file and not something else.

Thank you for your answer, I can confirm that the files I mostly use are `.txt` files, so I can see why the files have a different link formatting now, but I can also confirm that earlier versions did format links in `.txt` files as MD links.

In any case I think a global setting for `Insert link` when using share into file to make it enabled or disabled by default could be a good feature to add.

**Comment by harshad1** [comment]
Options are:

1. Remember the last used state to format links  <- I think I prefer this
2. Global option

I don't think that the per-file-format link syntax is not something that is easy to expose to the user

**Reply by gsantner** [reply]
introducing a option in settings menu UI sure is worse option

**Comment by harshad1** [comment]
Also note that we check the first and last word of every line to determine if we have links.

We can also share a file from markor into markor to link one file to another :)
